src/TypeScript/EslintBridgeClient/EslintBridgeProcess.cs

```diff
@@ -158,8 +158,18 @@ private void StartServer()
 
         private void Process_Exited(object sender, EventArgs e)
         {
-            logger.LogDebug($"[eslint-bridge] [process id: {processId}] Process exited.", processId);
-            ClearProcessData();
+            // Guard against a potential race condition i.e. this event being received
+            // after an new analysis request has already created a new Process
+            if(sender is Process terminatedProcess)
+            {
+                logger.LogDebug($"[eslint-bridge] [process id: {terminatedProcess.Id}] Process exited.");
+                terminatedProcess.Exited -= Process_Exited;
```

Comment:
This is the only place we unregister the event handler. In theory that should be enough.
The other code paths that call `ClearProcessData` will have disposed the old process so the event registration will have be removed.